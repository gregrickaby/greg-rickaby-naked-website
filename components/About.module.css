.about {
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  max-width: 740px;
  padding: 0 1.2rem;
}

@media (min-width: 768px) {
  .about {
    padding: 0;
  }
}
