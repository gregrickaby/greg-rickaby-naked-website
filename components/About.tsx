import styles from "./About.module.css";

export default function About() {
  return (
    <div id="about" className={styles.about}>
      <h2>About</h2>
      <p>
        Way back in 1997, I built my first website on Geocities and have been
        passionate about the web since! In my spare time I enjoy{" "}
        <a href="https://gregrickaby.smugmug.com" rel="author">
          taking photos
        </a>
        , cheering for the{" "}
        <a href="https://www.packers.com/">Green Bay Packers</a>, and making
        pizzas.
      </p>

      <h3>Web Developer</h3>
      <p>
        I&apos;m a Tech Lead at{" "}
        <a href="https://americaneagle.com/">AmericanEagle.com</a>, an industry
        leader in web design, development, hosting, and digital marketing. I
        partner with clients and team members to build websites with Next.js and
        WordPress.
      </p>
      <p>
        View my r√©sum√© on{" "}
        <a href="https://www.linkedin.com/in/gregrickaby/" rel="author">
          LinkedIn
        </a>
        .
      </p>

      <h3>Published Author</h3>
      <p>
        In 2017, I wrote a children&apos;s book titled{" "}
        <a href="https://amzn.to/34QAxAQ">
          <em>
            <strong>Creating a Website for Dummies Jr</strong>
          </em>
        </a>{" "}
        which was published by{" "}
        <a href="https://www.wiley.com/en-us"> Wiley Global</a> under the{" "}
        <a href="https://www.dummies.com/">dummies‚Ñ¢Ô∏è</a> brand.
      </p>

      <p>
        The book guides young readers through the planning and development
        processes of building a website. It has been translated into other 2
        other languages, and currently has a 4.5-star rating on Amazon.
      </p>

      <p>
        Since being published, I&apos;ve also served as the Technical Editor on
        three other books:
      </p>
      <ul>
        <li>
          <a href="https://amzn.to/37XCflU" rel="nofollow">
            <em>WordPress for Dummies</em>
          </a>{" "}
          (Sabin-Wilson, 2021)
        </li>
        <li>
          <a href="https://amzn.to/2BgCg7i" rel="nofollow">
            <em>Professional WordPress Plugin Development</em>
          </a>{" "}
          (Williams et al., 2020)
        </li>
        <li>
          <a href="https://amzn.to/37MMDLp" rel="nofollow">
            <em>WordPress All-In-One For Dummies</em>
          </a>{" "}
          (Sabin-Wilson, 2019)
        </li>
      </ul>

      <h3>Photographer</h3>
      <p>
        In late 2019, I bought a mirrorless camera and spent months taking
        courses, reading, practicing, and having my work critiqued by peers.
        What has started as an enjoyable hobby, has turned into a passionate way
        to express my creativity.
      </p>
      <p>
        View my{" "}
        <a href="https://gregrickaby.smugmug.com" rel="author">
          portfolio
        </a>{" "}
        on SmugMug.
      </p>

      <h3>Contributions</h3>
      <p>
        I've been part of the WordPress community since 2008. I've made
        contributions to core, documentation, plugins, and themes. I've also
        spoken at WordCamps and meet-ups. I'm also a contributor to other
        open-source projects like Next.js, Storybook, and Gatsby.
      </p>
      <p>
        Outside of tech, I volunteer my time and efforts at local community
        organizations. The Boy Scouts, Wiregrass Church, and the Enterprise High
        School Band Boosters to name a few.
      </p>

      <p>
        You can find me around the web with the links below. Stay safe out
        there, and thanks for reading!
        <span role="img" aria-label="cheers">
          üçª
        </span>
      </p>
    </div>
  );
}
