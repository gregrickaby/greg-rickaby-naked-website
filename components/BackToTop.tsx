'use client'

import {IconArrowUp} from '@tabler/icons-react'
import {useEffect, useState} from 'react'

/**
 * The back to top component.
 */
export default function BackToTop() {
  const [showButton, setShowButton] = useState(false)
  const buttonText = 'Go back to the top of the page'

  /**
   * Scroll handler.
   */
  const handleScroll = () => {
    window.scrollTo({
      top: 0,
      behavior: 'smooth'
    })
  }

  /**
   * Effect for showing the back to top button.
   */
  useEffect(() => {
    // Scroll event handler.
    const handleScroll = () => {
      setShowButton(window.scrollY > 200)
    }

    // Add event listener.
    window.addEventListener('scroll', handleScroll)

    // Cleanup event listener.
    return () => window.removeEventListener('scroll', handleScroll)
  }, [])

  return showButton ? (
    <button
      aria-label={buttonText}
      className="button fixed bottom-8 right-3 rounded-full bg-gray-400 p-2 text-white shadow-lg hover:bg-gray-500 lg:right-12 dark:bg-gray-800 hover:dark:bg-gray-700"
      data-testid="back-to-top"
      onClick={handleScroll}
      title={buttonText}
    >
      <IconArrowUp height="32" width="32" />
      <span className="sr-only">{buttonText}</span>
    </button>
  ) : null
}
