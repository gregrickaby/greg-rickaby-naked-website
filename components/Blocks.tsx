'use client'

import '@fancyapps/ui/dist/fancybox/fancybox.css'
import {useEffect, useRef} from 'react'

/**
 * The blocks component.
 */
export default function Blocks({content}: Readonly<{content: string}>) {
  // Create a ref for the post content.
  const contentRef = useRef<HTMLDivElement>(null)

  /**
   * Initialize photo gallery.
   */
  async function galleryInit() {
    // On the server? Bail.
    if (typeof window === 'undefined') {
      return
    }

    // No ref? Bail.
    if (!contentRef.current) {
      return
    }

    // Attempt to find the gallery.
    const gallery = contentRef.current.querySelector('.grd-photo-gallery-grid')

    // No gallery? Bail.
    if (!gallery) {
      return
    }

    // Import the libraries on demand.
    const Masonry = (await import('masonry-layout')).default
    const Fancybox = (await import('@fancyapps/ui')).Fancybox
    const imagesLoaded = (await import('imagesloaded')).default

    // Initialize ImagesLoaded.
    const imagesLoadedInstance = imagesLoaded(gallery)

    // Initialize Masonry.
    const masonryGrid = new Masonry(gallery, {
      gutter: 24,
      itemSelector: '.grd-photo-gallery-image',
      columnWidth: '.grd-photo-gallery-grid-sizer',
      percentPosition: true
    })

    // When ImagesLoaded is done, initialize Masonry.
    imagesLoadedInstance.on('done', () => {
      masonryGrid?.layout?.()
    })

    // Initialize Fancybox.
    Fancybox.bind('[data-fancybox]', {
      animated: false,
      groupAll: true,
      defaultDisplay: 'flex',
      Toolbar: {
        display: {
          left: ['infobar'],
          middle: [
            'zoomIn',
            'zoomOut',
            'toggle1to1',
            'rotateCCW',
            'rotateCW',
            'flipX',
            'flipY'
          ],
          right: ['slideshow', 'thumbs', 'download', 'close']
        }
      }
    })

    // Destroy Masonry and Fancybox when the component unmounts.
    return () => {
      masonryGrid?.destroy?.()
      Fancybox.destroy?.()
    }
  }

  /**
   * Initialize the gallery when the component mounts.
   */
  useEffect(() => {
    galleryInit()
  }, [content])

  return (
    <div
      ref={contentRef}
      className="text-lg"
      dangerouslySetInnerHTML={{__html: content}}
    />
  )
}
