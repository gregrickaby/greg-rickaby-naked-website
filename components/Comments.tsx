import CommentForm from '@/components/CommentForm'
import type {Post} from '@/lib/types'

/**
 * Comments component.
 */
export default function Comments({post}: Readonly<{post: Post}>) {
  return (
    <>
      <section className="border-t-2 border-zinc-500 py-4">
        <h3 className="text-3xl font-bold">Comments</h3>
        {!!post?.comments?.edges?.length && (
          <p className="italic">No comments yet.</p>
        )}
        {post.comments.edges.map((comment) => (
          <article key={comment.node.databaseId}>
            <header className="flex items-center gap-2">
              {!!comment?.node?.author?.node?.avatar?.url && (
                <img
                  alt={comment.node.author.node.name}
                  className="m-0 rounded-full"
                  height={64}
                  src={comment.node.author.node.avatar.url}
                  width={64}
                />
              )}
              <div className="flex flex-col gap-2">
                <h4
                  className="m-0 p-0 leading-none"
                  dangerouslySetInnerHTML={{
                    __html: comment.node.author.node.name
                  }}
                />
                <time className="italic">{comment.node.date}</time>
              </div>
            </header>
            <div dangerouslySetInnerHTML={{__html: comment.node.content}} />
          </article>
        ))}
      </section>
      <CommentForm postID={post.databaseId} />
    </>
  )
}
