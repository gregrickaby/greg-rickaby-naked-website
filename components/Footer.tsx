import SocialLinks from '@/components/SocialLinks'

/**
 * Footer component.
 */
export default function Footer() {
  return (
    <footer className="mx-auto w-full border-t border-zinc-500 pt-8 text-center text-sm">
      <SocialLinks />
      <p>
        Unless otherwise noted, all content on this website is licensed under{' '}
        <a href="https://creativecommons.org/licenses/by-nc-nd/4.0/">
          CC BY-NC-ND 4.0
        </a>
        .
      </p>

      <p>&copy; 2008 - {new Date().getFullYear()} Greg Rickaby</p>
    </footer>
  )
}
