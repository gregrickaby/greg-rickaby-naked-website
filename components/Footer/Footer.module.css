.footer {
  @apply mt-20 p-8 text-center text-sm md:p-12;
}
