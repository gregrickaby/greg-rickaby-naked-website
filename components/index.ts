export * from './BackToTop'
export * from './Blocks'
export * from './Footer'
export * from './Header'
export * from './JsonLd'
export * from './Navigation'
export * from './SocialLinks'
