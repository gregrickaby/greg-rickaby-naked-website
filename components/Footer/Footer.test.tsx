import {Footer} from '@/components/Footer'
import '@testing-library/jest-dom'
import {render} from '@testing-library/react'
import {axe} from 'jest-axe'
import {describe, expect, it, vi} from 'vitest'

/**
 * Mock the <SocialLinks> component.
 */
vi.mock('@/components/SocialLinks', () => ({
  SocialLinks: function DummySocialLinks() {
    return <div data-testid="social-links"></div>
  }
}))

/**
 * Test suite for the Footer component.
 */
describe('Footer', () => {
  it('should render', () => {
    const {getByTestId} = render(<Footer />)
    const footer = getByTestId('footer')
    expect(footer).toBeInTheDocument()
  })

  it('should render the SocialLinks component', () => {
    const screen = render(<Footer />)
    const socialLinks = screen.getByTestId('social-links')
    expect(socialLinks).toBeInTheDocument()
  })

  it('should display the correct copyright year', () => {
    const {getByText} = render(<Footer />)
    const currentYear = new Date().getFullYear()
    expect(
      getByText(`© 2008 - ${currentYear} Greg Rickaby ✌️`)
    ).toBeInTheDocument()
  })

  it('should display the correct license information', () => {
    const screen = render(<Footer />)
    const licenseLink = screen.getByRole('link', {
      name: /CC BY-NC-ND 4.0/i
    })
    expect(licenseLink).toBeInTheDocument()
    expect(licenseLink).toHaveAttribute(
      'href',
      'https://creativecommons.org/licenses/by-nc-nd/4.0/'
    )
  })

  it('should not have any accessibility issues', async () => {
    const {container} = render(<Footer />)
    const results = await axe(container)
    expect(results).toHaveNoViolations()
  })

  it('should match snapshot', () => {
    const {getByTestId} = render(<Footer />)
    const footer = getByTestId('footer')
    expect(footer).toMatchSnapshot()
  })
})
