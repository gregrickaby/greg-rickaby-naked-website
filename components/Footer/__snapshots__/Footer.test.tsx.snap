// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > should match snapshot 1`] = `
<footer
  class="_footer_3059f1"
  data-testid="footer"
>
  <div
    data-testid="social-links"
  />
  <p>
    Unless otherwise noted, all content on this website is licensed under
     
    <a
      href="https://creativecommons.org/licenses/by-nc-nd/4.0/"
    >
      CC BY-NC-ND 4.0
    </a>
    .
  </p>
  <p>
    © 2008 - 
    2024
     Greg Rickaby ✌️
  </p>
</footer>
`;
